refextract/references/pdf.py

```diff
@@ -46,8 +49,12 @@ def extract_texkeys_from_pdf(pdf_file):
     @return: list of all texkeys found in the PDF
     """
     with open(pdf_file, 'rb') as pdf_stream:
-        pdf = PdfFileReader(pdf_stream, strict=False)
-        destinations = pdf.getNamedDestinations()
+        try:
+            pdf = PdfFileReader(pdf_stream, strict=False)
+            destinations = pdf.getNamedDestinations()
+        except PyPdfError as exc:
+            print("* PDF: Internal PyPDF2 error, no TeXkeys returned.", exc)
```

Comment:
In general, say libraries should never print, since they are going not to be used in an interactive way. `refextract` here is used as a sort of library, since the client code will be the workflow.

On the other hand if one has to communicate an error and can't use logging (like in this case, since it would require some refactoring), then it's better to at least print on stderr, where errors are due.

Also see: http://blog.dscpl.com.au/2009/04/wsgi-and-printing-to-standard-output.html